src/AppBundle/Validator/Constraints/UniquePropertyValidator.php

```diff
@@ -0,0 +1,138 @@
+     * @throws UnexpectedTypeException       If the ExecutionContext is invalid (to be removed when upgrading to 3.0, just used in order to verify the correct context).
+     * @throws UnexpectedTypeException       If the value is not a scalar value or has a __toString() interceptor.
+     * @throws ConstraintDefinitionException If the manager alias could not be loaded or there's no manager for a specific class.
+     * @throws ConstraintDefinitionException If the entity field cannot be found.
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        /** @var UniqueProperty $constraint */
+        if (!$constraint instanceof UniqueProperty) {
+            throw new UnexpectedTypeException($constraint, UniqueProperty::class);
+        }
+
+        /** @var ExecutionContextInterface $context */
+        $context = $this->context;
+        if (!$context instanceof ExecutionContextInterface) {
+            throw new UnexpectedTypeException($context, ExecutionContextInterface::class);
+        }
+
+        if (!is_scalar($value)
+            && (!is_object($value) && !method_exists($value, '__toString'))
+        ) {
+            throw new UnexpectedTypeException($value, 'scalar');
+        }
+
+        $value       = (string) $value;
+        $entityAlias = $constraint->entity;
+
+        if (null !== $managerAlias = $constraint->manager) {
+            $manager = $this->managerRegistry->getManager($managerAlias);
+
+            if (!$manager) {
+                throw new ConstraintDefinitionException(sprintf('No such entity manager with alias "%s"!', $managerAlias));
+            }
+        } else {
+            $manager = $this->managerRegistry->getManagerForClass($entityAlias);
+
+            if (!$manager) {
+                throw new ConstraintDefinitionException(sprintf('Cannot find entity manager for model "%s"!', $entityAlias));
+            }
+        }
+
+        $field = $constraint->field;
+        /** @var \Doctrine\ORM\Mapping\ClassMetadata $metadata */
+        $metadata = $manager->getClassMetadata($entityAlias);
+
+        if (!$metadata->hasField($field) && !$metadata->hasAssociation($field)) {
+            if (false !== strpos($field, '.')) {
+                list($embeddable, $embeddedField) = explode('.', $field, 2);
+
+                if (isset($metadata->embeddedClasses[$embeddable])) {
+                    /** @var \Doctrine\ORM\Mapping\ClassMetadata $embeddableMetadata */
+                    $embeddableMetadata = $manager->getClassMetadata($metadata->embeddedClasses[$embeddable]['class']);
+
+                    if (!$embeddableMetadata->hasField($embeddedField)) {
+                        throw new ConstraintDefinitionException(
+                            sprintf('Embeddable "%s" on entity "%s" has no field "%s"!', $embeddable, $entityAlias, $embeddedField)
+                        );
+                    }
+                } else {
```

Comment:
this block can be reached when checking a value of a OneToOne relation, so the exception may confuse